ush/cdo_post_ocean_format.sh

```diff
@@ -0,0 +1,193 @@
+#
+# Description:
+#   This function takes an input string and removes all whitespace
+#   characters (spaces, tabs, and newline characters) to produce a
+#   cleaned output string.
+#
+# Parameters:
+#   $1 - The input string from which whitespace will be removed.
+#
+# Return:
+#   The cleaned string with no whitespace.
+#
+# Example usage:
+#   cleaned_string=$(_strip_whitespace "   This is a string   with spaces  ")
+#   echo "Cleaned string: \"$cleaned_string\""
+#
+# This example will remove all leading, trailing, and internal
+# whitespace from the input string and display the cleaned result.
+function _strip_whitespace(){
+    local in_string="${1}"
+
+    out_string=$(echo "${in_string}" | $(command -v sed) "s/ //g")
+}
+
+#######
+
+# ncattr_update - Update/add attributes for a variable in a netCDF file.
+#
+# Description:
+#   This function updates the specified attribute for a specified
+#     variable in a netCDF file using the `ncatted` command.
+#
+# Parameters:
+#   $1 - The variable name to update.
+#   $2 - netCDF variable metadata attribute name.
+#   $3 - The coordinates as a comma-separated string.
+#
+# Global Variables:
+#   global_array - An array containing the split coordinates.
+#   output_path - The path to the output netCDF file.
+#
+# Example usage:
+#   ncupdate "variable_name" "coords" "lon,lat,time"
+#
+# This example updates the `coords` attributes for the specified
+#   variable and writes the updates to the output netCDF file.
+function ncattr_update(){
+    local varname="${1}"
+    local ncattr="${2}"
+    local coords="${3}"
+
+    _comma_split_string "${coords}"
+    coords="${global_array[@]}"
+    coords_str="$(echo "${coords}" | $(command -v tr) -s ' ')"
+    ncattr_str="$(echo "${ncattr}" | $(command -v tr) -s ' ')"
+    echo "Adding netCDF attribute ${ncattr_str} values ${coords_str} to variable ${varname} metadata and writing to file ${output_path}"
+    ($(command -v ncatted) -O -a "${ncattr_str}","${varname}",c,c," ${coords_str}" "${output_path}" "${output_path}")
+}
+
+#######
+
+start_time=$(date +%s)
+_calling_script=$(basename "${BASH_SOURCE[0]}")
+start_time_human=$(date -d"@${start_time}" -u)
+echo "Begin ${_calling_script} at ${start_time_human}."
+
+# Copy the input file path to the output file path.
+echo "Copying file ${input_path} to ${output_path} and preparing for variable updates."
+$(command -v cp) "${input_path}" "${output_path}"
+
+# Read the configuration file for the the variables to be updated and
+# proceed accordingly.
+while IFS= read -r line; do
+
+    # Get the attributes for the respective variable.
+    varname=$(echo "${line}" | $(command -v awk) '{print $1}')
+    ncattr=$(echo "${line}" | $(command -v awk) '{print $2}')
+    coords=$(echo "${line}" | $(command -v awk) '{print $3}')
+    
+    # Update the variable attributes and write the updates to the
+    # specified output file (see `output_path`).
+    ncattr_update "${varname}" "${ncattr}" "${coords}"
```

Comment:
Can't this just be:
```suggestion
    # shellcheck disable=SC2086
    ncattr_update ${line}
```
(The shellcheck directive is to stifle complaints about the lack of quotation marks.)

Reply:
This does not guarantee that the string will be parsed properly into the attributes that are required. The `awk` issue can be eliminated by adding an exception to the shell linter. Any use of `awk` will fail with the current configuration.

Reply:
I wasn't worried about the linter so much as just on a crusade to eliminate as many calls to external programs like `sed` and `tr` as possible for simple functions. I still don't see the difference, but I trust you. If you do need to use `sed`, prefer herestrings (`<<< ${line}`) over echoing through a pipe.

Reply:
OK, I will give this another look. It's possible that I overlooked something previously.